Adding config task

roelsgaard opened the pull request:
Signed-off-by: Chris Roelsgaard Rasmussen <q1a7@nykredit.dk>

roelsgaard commented:
This task will create an angular constant containg the content from the app/config.json (or the file defined in gulp.config.app.configFile).
This constant can then be used in directives and services as e.x configurable service endpoints.

jotka commented on README.md:
```diff
@@ -249,7 +252,8 @@ The `gulpfile.js` from Gript contains also these specific tasks:
 - **mocks** : starts a server with mock services. Refer to the [Mock server](#mocks) section for guidelines.
-- **modernizr** : builds custom [Modernizr](http://modernizr.com) script and injects it into `index.html`  
+- **modernizr** : builds custom [Modernizr](http://modernizr.com) script and injects it into `index.html`
+- **config** : creates an angular constant for the module specified in `gulp.config.module` with the content of the json file placed at `app/config.json` or configured in `gulp.config.app.configFile`   
```
1. we need an example config.json in the package/sample_configs directory
2. we need the app.configFile section in the package/sample_configs/gulpfile.js
3. there is a new section "app" now in the gulpfile.js, so I think we should throw also the "module" into that section and then update gulpfile.js and  package/sample_configs/gulpfile.js

jotka commented on tasks/build.gulp.js:
```diff
@@ -133,7 +133,7 @@ module.exports = function (gulp) {
     });
 
-    gulp.task('build', ['version', 'inject', 'images', 'fonts', 'resources', 'lib', 'lint-js'], function (callback) {
+    gulp.task('build', ['config', 'version', 'inject', 'images', 'fonts', 'resources', 'lib', 'lint-js'], function (callback) {
```
'config' generates the JavaScript file which is then being injected into the index.html. Therefore it must not be executed in parallel with 'inject'. It will probably never happen, but I would recommend to execute it in the 'inject' section, like 'modernizr' or something.

jotka commented:
hi @roelsgaard please try to build the project before submitting a pull-request. There is a bunch of linter errors in config.gulp.js. I will correct them.

`tasks\config.gulp.js
   3:26  error    Missing space before function parentheses      space-before-function-paren
   3:32  warning  Missing space before opening brace             space-before-blocks
   4:21  warning  Multiple spaces found before '='               no-multi-spaces
   5:1   error    Trailing spaces not allowed                    no-trailing-spaces
   8:1   error    Trailing spaces not allowed                    no-trailing-spaces
  15:2   error    Newline required at end of file but not found  eol-last`

roelsgaard commented:
hi @jotka 
I have now addressed you issues and updated the pullrequest.
Sorry for the failing lints and missing implementations in the package section - i now have a lot better understanding of the setup :-)
